doc/src/advanced_tutorial/add_move_minimal.md

```diff
@@ -0,0 +1,236 @@
+
+```rust
+// File: translate.rs
+impl Translate {
+    fn new(dr: f64) -> Translate {
+        Translate {
+            old_system: System::default(),
+            dr: dr,
+            range: Range::new(-dr, dr),
+        }
+    }
+}
+```
+
+The `new` function takes a single argument, the maximum displacement `dr`.
+The internal unit of length - the displacement is a length - is Angstrom.
+Remember that in rust, we need to initialize all fields of our struct, so we use the default implementation for our system.
+
+```rust
+impl Default for Translate {
+    fn default() -> Translate {
+        Translate::new(1.0)
+    }
+}
+``` 
+
+We also implement `Default`, which simply calls `new` with a displacement of `dr = 1.0 angstrom`. 
+
+## Implement the `MCMove` trait.
+
+Good. We are able to create an instance of the struct.
+Let's start implementing the `MCMove` trait.
+All functions that follow are within the `impl` brackets.
+
+```rust
+impl MCMove for Translate {
+    // here will be our functions.
+}
+```
+
+First, we implement the `describe` function:
+
+```rust
+fn describe(&self) -> &str {
+    "molecular translation"
+}
+```
+
+Pretty self explanatory - a simple description of what the move does.
+
+Next, the `setup`.
+As mentioned earlier, this function is called once before the simulations starts.
+Since it takes the system as parameter, we can compute quantities that may be important for the move - depending on the system we use.
+For our initial implementation though, we don't use the `setup` function.
+
+```rust
+fn setup(&mut self, _: &System) { 
+    // For now, do nothing.
+}
+```
+The `prepare` function is a bit more complicated:
+
+```rust
+fn prepare(&mut self, system: &mut System, rng: &mut Box<Rng>) -> bool {
+    // select a molecule randomly: pick a random number between 0 and the total number
+    // of molecules in the system.
+    let nmols = system.molecules().len();
+    if nmols == 0 {
+        return false // no molecule, so we can't perform the move
+    } else {
+        let molid = rng.gen_range(0, nmols)
+    }
+
+    // create a vector, containing the random displacements in every cartesian
+    // direction.
+    let delta = Vector3D::new(
+        self.range.sample(rng), // x
+        self.range.sample(rng), // y
+        self.range.sample(rng), // z
+    )
+
+    self.old_system = system.clone(); // before modifying the system, create a copy
+    // use the current positions and add the displacement
+    for i in self.old_system.molecule(molid) {
+        system[i].position += delta
+    }
+    true // we set everything up, so we can perform the move.
+}
+```
+
+That is a lot of stuff.
+Let's first have a look at the function arguments:
+We take three mutable references: One to the `Translate` struct (`&mut self`), one
+to the system and finally one to `Box<Rng>`, which is a **r**andom **n**umber **g**enerator.
+
+Let us now go through the code blocks one by one.
```

Comment:
You could add markers `(1)`, `(2)`, `(3)` in the comments and refer to them here.